### PR TITLE
Parvathy, Rahul | GOK-177 | Add. Abnormal Checkbox For All Data Types

### PR DESCRIPTION
**Pull request by rahu1ramesh**
**Add. Abnormal Checkbox For All Data Types**

- [x] This PR has a proper title that briefly describes the work done
- [x] I have squashed / amended the comments to make it more redable
- [x] I have included link to all the JIRA ticket('s)
- [x] I wrote the code as a pair or atleast performed a self-review of my own code
- [x] I have updated the documentation on [Bahmni Wiki](https://bahmni.atlassian.net/wiki/spaces/BAH/overview) or [README](https://github.com/Bahmni/bahmni-lab-frontend/blob/main/README.md) (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary
In this PR, abnormal checkbox is added to all the data types when entering the lab results. The following requirements are met :
 - Add abnormal check box for all data type on lablite
 - If the test result deviates from normal range for numeric data type, “check”box should automatically selected
 - For other than non numeric data type the user should be able to select the abnormal checkbox manually
 - If abnormal box is checked show the result on red colour.
 - One can manually uncheck the abnormal option.When it manually uncheck remove the red colour from test results.

## Screenshots
Screenshot 2023-07-10 at 3 45 48 PM
Screenshot 2023-07-10 at 3 48 06 PM

## JIRA tickets
Trello -> https://trello.com/c/v3cYXB9r

### CODE REVIEW

File: src/patient-lab-dashboard/test-results/test-results.test.tsx [hunk 1]
```diff
@@ -168,7 +168,7 @@ describe('TestResults Report', () => {
       color: 'red',
     })
   })
-  it('should indicate when the entered value is abnormal based on Checkbox', async () => {
+  it('te', async () => {
```
Comment (abinaya-u) [inline src/patient-lab-dashboard/test-results/test-results.test.tsx]: can we revert back this test name

File: src/patient-lab-dashboard/test-results/test-results.tsx [hunk 2]
```diff
@@ -216,27 +216,41 @@ const TestResults: React.FC<TestResultProps> = ({
...
+      return (
+        <div className={styles.inputFieldWithCheckbox}>
+          {datatype === 'Boolean' || datatype === 'Coded' ? (
             <Dropdown
               titleText={getTestNameWithUnits(test)}
               id="answers-list-dropdown"
               title="answers list"
               items={items}
-              itemToString={data => data.name.name}
+              itemToString={getDropdownItemToString(test, labResult)}
```
Comment (abinaya-u) [inline src/patient-lab-dashboard/test-results/test-results.tsx]: passing labResult as a parameter is not required since it is a state variable

Comment (rahu1ramesh) [reply src/patient-lab-dashboard/test-results/test-results.tsx]: The parameter list for getDropdownItem has been updated, thanks @abinaya-u for the suggestion.

File: src/patient-lab-dashboard/test-results/test-results.test.tsx (hunk 1 above)
Comment (rahu1ramesh) [reply src/patient-lab-dashboard/test-results/test-results.test.tsx]: The test name has been reverted back to 'should indicate when the entered value is abnormal based on Checkbox'.

**Review by abinaya-u — APPROVED** [review]

**Pull request merged**